packages/cli/src/util/dev/server.ts

```diff
@@ -2641,15 +2633,29 @@ function needsBlockingBuild(buildMatch: BuildMatch): boolean {
   return typeof builder.shouldServe !== 'function';
 }
 
-async function checkForPort(port: number, timeout: number): Promise<void> {
-  const opts = { host: '127.0.0.1' };
+async function checkForPort(port: number, timeout: number): Promise<string> {
+  let host;
   const start = Date.now();
-  while (!(await isPortReachable(port, opts))) {
+  while (!(host = await getReachableHostOnPort(port))) {
     if (Date.now() - start > timeout) {
-      throw new Error(`Detecting port ${port} timed out after ${timeout}ms`);
+      break;
     }
     await sleep(100);
   }
+  if (!host) {
+    throw new Error(`Detecting port ${port} timed out after ${timeout}ms`);
+  }
+  return host;
+}
+
+async function getReachableHostOnPort(port: number): Promise<string | false> {
+  const optsIpv4 = { host: '127.0.0.1' };
+  const optsIpv6 = { host: '::1' };
+  const results = await Promise.all([
+    isPortReachable(port, optsIpv4).then(r => r && optsIpv4.host),
+    isPortReachable(port, optsIpv6).then(r => r && `[${optsIpv6.host}]`),
+  ]);
+  return results.find(Boolean) || false;
```

Comment:
What about if both are found? Looks like this will favor ipv4 over ipv6, is that intended?

Reply:
If both are reachable then it shouldn't really matter. But we could swap to prefer IPv6.

Reply:
I think since Node.js prefers ipv6 by default, we probably should too